core/src/main/java/org/everit/json/schema/loader/ObjectSchemaLoader.java

```diff
@@ -29,38 +31,43 @@ public ObjectSchemaLoader(LoadingState ls, LoaderConfig config, SchemaLoader def
         ls.schemaJson().maybe("minProperties").map(JsonValue::requireInteger).ifPresent(builder::minProperties);
         ls.schemaJson().maybe("maxProperties").map(JsonValue::requireInteger).ifPresent(builder::maxProperties);
         ls.schemaJson().maybe("properties").map(JsonValue::requireObject)
-                .ifPresent(propertyDefs -> populatePropertySchemas(propertyDefs, builder));
+            .ifPresent(propertyDefs -> populatePropertySchemas(propertyDefs, builder));
         ls.schemaJson().maybe("additionalProperties").ifPresent(rawAddProps -> {
             rawAddProps.canBe(Boolean.class, p -> builder.additionalProperties(p))
-                    .or(JsonObject.class, def -> builder.schemaOfAdditionalProperties(defaultLoader.loadChild(def).build()))
-                    .requireAny();
+                .or(JsonObject.class, def -> builder.schemaOfAdditionalProperties(defaultLoader.loadChild(def).build()))
+                .requireAny();
         });
         ls.schemaJson().maybe("required").map(JsonValue::requireArray)
-                .ifPresent(arr -> arr.forEach((i, val) -> builder.addRequiredProperty(val.requireString())));
+            .ifPresent(arr -> arr.forEach((i, val) -> builder.addRequiredProperty(val.requireString())));
         ls.schemaJson().maybe("patternProperties").map(JsonValue::requireObject)
-                .ifPresent(patternProps -> {
-                    patternProps.keySet().forEach(pattern -> {
-                        Schema patternSchema = defaultLoader.loadChild(patternProps.require(pattern)).build();
-                        Regexp regexp = ls.config.regexpFactory.createHandler(pattern);
-                        builder.patternProperty(regexp, patternSchema);
-                    });
+            .ifPresent(patternProps -> {
+                patternProps.keySet().forEach(pattern -> {
+                    Schema patternSchema = defaultLoader.loadChild(patternProps.require(pattern)).build();
+                    Regexp regexp = ls.config.regexpFactory.createHandler(pattern);
+                    builder.patternProperty(regexp, patternSchema);
                 });
-        ls.schemaJson().maybe("dependencies").map(JsonValue::requireObject)
+            });
+        if (ls.specVersion().isAtLeast(DRAFT_201909)) {
+            ls.schemaJson().maybe("dependentRequired").map(JsonValue::requireObject)
+                .ifPresent(deps -> addDependencies(builder, deps));
+        } else {
+            ls.schemaJson().maybe("dependencies").map(JsonValue::requireObject)
                 .ifPresent(deps -> addDependencies(builder, deps));
+        }
```

Comment:
Instead of handling it here, I think it would be more consistent to add a `String dependentRequiredKeyword()` abstract method to the `SpecificationVersion` enum, which returns `"dependentRequired"` for draft201909,  and `"required"` for all the older enum values. I suggest this because `SpecificationVersion#idKeyword()` serves a similar purpose, so it would be a consistent approach.

Reply:
Cool, I did not notice `SpecificationVersion#idKeyword()`